packages/flutter_tools/lib/src/isolated/native_assets/android/native_assets.dart

```diff
@@ -189,36 +186,37 @@ AndroidArch _getAndroidArch(Target target) {
   }
 }
 
-Map<Asset, KernelAsset> _assetTargetLocations(List<Asset> nativeAssets) {
-  return <Asset, KernelAsset>{
-    for (final Asset asset in nativeAssets)
+Map<AssetImpl, KernelAsset> _assetTargetLocations(
+    List<AssetImpl> nativeAssets) {
+  return <AssetImpl, KernelAsset>{
+    for (final AssetImpl asset in nativeAssets)
       asset: _targetLocationAndroid(asset),
   };
 }
 
 /// Converts the `path` of [asset] as output from a `build.dart` invocation to
 /// the path used inside the Flutter app bundle.
-KernelAsset _targetLocationAndroid(Asset asset) {
-  final AssetPath path = asset.path;
+KernelAsset _targetLocationAndroid(AssetImpl asset) {
+  final LinkModeImpl linkMode = (asset as NativeCodeAssetImpl).linkMode;
```

Comment:
Out of curiosity, why is the tool now using *Impl types? I suppose since we're already importing an *internal library it doesn't matter also using internal implementation interfaces.

Reply:
The interfaces don't expose the APIs needed. (JSON serialization for example.) That's why we use the implementations.

The API is meant for `hook/built.dart` writers. The `*Impl` classes are meant for `hook/build.dart` invokers (`dartdev` and `flutter_tools`).